src/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.Executor.cs

```diff
@@ -159,78 +134,6 @@ await DocumentAnalysisExecutor.CreateCompilationWithAnalyzersAsync(
             }
         }
 
-        private static ImmutableDictionary<DiagnosticAnalyzer, DiagnosticAnalysisResult> MergeExistingDiagnostics(
-            VersionStamp version, ImmutableDictionary<DiagnosticAnalyzer, DiagnosticAnalysisResult> existing, ImmutableDictionary<DiagnosticAnalyzer, DiagnosticAnalysisResult> result)
-        {
-            // quick bail out.
-            if (existing.IsEmpty)
-            {
-                return result;
-            }
-
-            foreach (var (analyzer, results) in existing)
-            {
-                if (results.Version != version)
-                {
-                    continue;
-                }
-
-                result = result.SetItem(analyzer, results);
-            }
-
-            return result;
-        }
-
-        private static bool TryReduceAnalyzersToRun(
-            CompilationWithAnalyzersPair compilationWithAnalyzers, VersionStamp version,
-            ImmutableDictionary<DiagnosticAnalyzer, DiagnosticAnalysisResult> existing,
-            out ImmutableArray<DiagnosticAnalyzer> projectAnalyzers,
-            out ImmutableArray<DiagnosticAnalyzer> hostAnalyzers)
-        {
-            projectAnalyzers = compilationWithAnalyzers.ProjectAnalyzers.WhereAsArray(
-                static (analyzer, arg) =>
-                {
-                    if (arg.existing.TryGetValue(analyzer, out var analysisResult) &&
```

Comment:
the crux to realize here is that 'existing' was never present (since it comes from the last cached data when ForceANalyzeProjectAsync was called, and that is not called unless user explicitly runs 'run code analysis').

And, if the user does that *and* the project is the same, then a higher layer reuses the cached results and we don't get here now.  So this would always end up returning 'true' in this lambda (and hte lambda below).WHich would cause the arrays to stay the same length in the lowest if check in this method.  leading us to return 'false' for the entire method.  So the caller in the method above can delete all this unnecessary code.

Reply:
> then a higher layer reuses the cached results and we don't get here now.

this being the new cwt with `Project` keys?